File: ProduitWebMobile/src/Entity/CategorieProduit.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;


/**
 * CategorieProduit
 *
 * @ORM\Table(name="categorie_produit", uniqueConstraints={@ORM\UniqueConstraint(name="idCategorie", columns={"idCategorie"})})
 * @ORM\Entity
 */
class CategorieProduit
{
    /**
     * @var int

     * @ORM\Column(name="idCategorie", type="integer", nullable=false)
     * @ORM\Id
     * @Groups ("post:read")
     */
    private $idcategorie;

    /**
     * @var string
     *
     * @ORM\Column(name="typeCategorie", type="string", length=255, nullable=false)

     * @ORM\GeneratedValue(strategy="IDENTITY")
     * @Groups ("post:read")
     */
    private $typecategorie;

    /**
     * @var string
     *
     * @ORM\Column(name="Description", type="string", length=255, nullable=false)
     * @Groups ("post:read")
     */
    private $description;

    /**
     * @var string|null
     *
     * @ORM\Column(name="image", type="string", length=255, nullable=true)
     * @Groups ("post:read")
     */
    private $image;

    /**
     * @return int
     */
    public function getIdcategorie(): ?int
    {
        return $this->idcategorie;
    }

    /**
     * @param int $idcategorie
     */
    public function setIdcategorie(int $idcategorie): void
    {
        $this->idcategorie = $idcategorie;
    }

    /**
     * @return string
     */
    public function getTypecategorie(): ?string
    {
        return $this->typecategorie;
    }

    /**
     * @param string $typecategorie
     */
    public function setTypecategorie(string $typecategorie): void
    {
        $this->typecategorie = $typecategorie;
    }

    /**
     * @return string
     */
    public function getDescription(): ?string
    {
        return $this->description;
    }

    /**
     * @param string $description
     */
    public function setDescription(string $description): void
    {
        $this->description = $description;
    }

    /**
     * @return string|null
     */
    public function getImage(): ?string
    {
        return $this->image;
    }

    /**
     * @param string|null $image
     */
    public function setImage(?string $image): void
    {
        $this->image = $image;
    }

    public function __toString()
    {
        return $this->getTypecategorie();
    }

}

File: mobilefinaleintegration/src/com/mycompany/myapp/entities/CategorieProduit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

/**
 *
 * @author user
 */
public class CategorieProduit {
    private String typecategorie;
    private String description;
    private String image;
    private  int idcategorie;

    public CategorieProduit(String typecategorie, String description, String image, int idcategorie) {
        this.typecategorie = typecategorie;
        this.description = description;
        this.image = image;
        this.idcategorie = idcategorie;
    }

    public CategorieProduit() {
    }

    public String getTypecategorie() {
        return typecategorie;
    }

    public void setTypecategorie(String typecategorie) {
        this.typecategorie = typecategorie;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public int getIdcategorie() {
        return idcategorie;
    }

    public void setIdcategorie(int idcategorie) {
        this.idcategorie = idcategorie;
    }

    @Override
    public String toString() {
        return "CategorieProduit{" + "typecategorie=" + typecategorie + ", description=" + description + ", image=" + image + ", idcategorie=" + idcategorie + '}';
    }
    

}
